Utils/Editor/FindUnusedAssetsWindow.cs

```diff
@@ -0,0 +1,241 @@
+							if (GUI.Button(new Rect(260, positionY, 70, EditorGUIUtility.singleLineHeight), "Select"))
+							{
+								Selection.activeObject = asset.Obj;
+							}
+							if (GUI.Button(new Rect(330, positionY, 70, EditorGUIUtility.singleLineHeight), "Delete"))
+							{
+								AssetDatabase.DeleteAsset(asset.Path);
+							}
+						}
+						else
+						{
+							GUI.Label(new Rect(10, positionY, 250, EditorGUIUtility.singleLineHeight), asset.Path);
+						}
+						GUILayout.EndHorizontal();
+					}
+				}
+				GUI.EndScrollView();
+			}
+		}
+
+		private static IEnumerable<Asset> FindUnusedObjects(IEnumerable<string> assetList, ICollection<string> usedAssets)
+		{
+			return (from assetPath in assetList
+				where !string.IsNullOrEmpty(assetPath) && !IsEditorPath(assetPath) && !usedAssets.Contains(assetPath)
+				select new Asset(assetPath, AssetDatabase.LoadAssetAtPath(assetPath, typeof(Object))))
+				.ToList();
+		}
+
+		private static bool IsEditorPath(string path)
+		{
+			return path.IndexOf("/editor/", StringComparison.CurrentCultureIgnoreCase) > 0;
+		}
+
+		private static string[] FindAllProjectAssets()
+		{
+			// Find all assets in the project
+			string[] tmpAssets1 = Directory.GetFiles(Application.dataPath, "*.*", SearchOption.AllDirectories);
+			// Filter assets
+			string[] tmpAssets2 = FilterAssets(tmpAssets1);
+			// Remove scene files to create the final list of assets.
+			string[] allAssets = Array.FindAll(tmpAssets2, name => !name.EndsWith(".unity"));
+
+			// Convert directory paths to asset paths
+			for (var i = 0; i < allAssets.Length; i++)
+			{
+				allAssets[i] = allAssets[i].Substring(allAssets[i].IndexOf("/Assets", StringComparison.Ordinal) + 1);
+				allAssets[i] = allAssets[i].Replace(@"\", "/");
+			}
+
+			return allAssets.ToArray();
+		}
+
+		private static string[] FilterAssets(string[] assets)
+		{
+			var filteredAssets = new List<string>();
+			foreach (var asset in assets)
+			{
+				var isValid = !asset.EndsWith(".meta");
+
+				if (!includeDuck)
+				{
+					isValid &= !asset.Contains("DUCK");
+				}
+				if (!includePlugins)
+				{
+					isValid &= !asset.Contains("Plugins");
+				}
+				if (!includeVendor)
+				{
+					isValid &= !asset.Contains("Vendor");
+				}
+
+				if (isValid)
+				{
+					filteredAssets.Add(asset);
+				}
+			}
+			return filteredAssets.ToArray();
+		}
+
+		private static List<string> FindUsedAssets()
+		{
+			var usedAssets = new List<string>();
+
+			try
+			{
+				var fileStream = new FileStream(GetEditorLogPath(), FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+				var streamReader = new StreamReader(fileStream);
+
+				var foundAssetsList = false;
+				var reachedEndOfAssetList = false;
+				while (!streamReader.EndOfStream && !foundAssetsList)
+				{
+					var line = streamReader.ReadLine();
+					if (line != null && line.Contains("Used Assets and files from the Resources folder, sorted by uncompressed size:"))
```

Comment:
This could change in each unity version. It's probably unlikely, but if it didn't find it, it's probably gonna look like nothing was unused. We may wanna throw if we don't find it,

Reply:
True, it has changed a couple of times in the last year already. I will add a way to handle this

Reply:
That would be a good thing. Not sure if there is any API for it, but in the worst case you can check the contents of the ProjectVersion.txt file in the ProjectSettings folder.

Reply:
Nevermind, just ignore this :D I got it wrong.